### PR TITLE
[eBPF] Adjust eBPF data reassembly

### PR DESCRIPTION
**Pull request by yinjiping**
We will mark the data reassembly push. If the inference is successful, we mark it as `MSG_REASM_START`. If the inference fails, we mark it as `MSG_REASM_SEG`.

There are two scenarios (below are data in the same direction with protocol reassembly enabled):

1. With both successfully inferred and failed inferred data:

```
request1 ------------------- (inference successful) ------- marked as: MSG_REASM_START
  request1_seg_1 ----------- (inference failed) ----------- marked as: MSG_REASM_SEG
  request1_seg_2 ----------- (inference failed) ----------- marked as: MSG_REASM_SEG
  request1_seg_3 ----------- (inference failed) ----------- marked as: MSG_REASM_SEG

```

2. Only successfully inferred data:

```
request1 ----------- (inference successful) --------------------- marked as: MSG_REASM_START
request2 ----------- (inference successful) --------------------- marked as: MSG_REASM_START
request3 ----------- (inference successful) --------------------- marked as: MSG_REASM_START
request4 ----------- (inference successful) --------------------- marked as: MSG_REASM_START

```

### This PR is for:

- Agent

#### Affected branches
- main
- v6.5

### CODE REVIEW

File: agent/src/ebpf/mod.rs [hunk 1]
```diff
@@ -146,6 +146,12 @@ pub const MSG_RESPONSE: u8 = 2;
 pub const MSG_REQUEST_END: u8 = 3;
 #[allow(dead_code)]
 pub const MSG_RESPONSE_END: u8 = 4;
+// The start of data reassembly.
+#[allow(dead_code)]
+pub const MSG_REASM_START: u8 = 5;
+// The segment of data reassembly.
+#[allow(dead_code)]
+pub const MSG_REASM_SEG: u8 = 6;
```
Comment (yinjiping) [inline agent/src/ebpf/mod.rs:154]: socket data 'msg_type', 添加了两种类型 MSG_REASM_START和MSG_REASM_SEG，来表示重组开始和重组段。
@yuanchaoa

File: agent/src/ebpf/samples/rust/profiler/Cargo.toml [hunk 2]
```diff
@@ -9,6 +9,7 @@ build = "build.rs"
 dunce = "0.1.1"
 
 [dependencies]
+cfg-if = "1.0"
```
Comment (yuanchaoa) [inline agent/src/ebpf/samples/rust/profiler/Cargo.toml:12]: 这个目录看没有代码修改 ， 加错了 ？

File: agent/src/ebpf/samples/rust/socket-tracer/Cargo.toml [hunk 3]
```diff
@@ -9,6 +9,7 @@ build = "build.rs"
 dunce = "0.1.1"
 
 [dependencies]
+cfg-if = "1.0"
```
Comment (yuanchaoa) [inline agent/src/ebpf/samples/rust/socket-tracer/Cargo.toml:12]: 同上

File: agent/src/ebpf/samples/rust/profiler/Cargo.toml (hunk 2 above)
Comment (yinjiping) [reply agent/src/ebpf/samples/rust/profiler/Cargo.toml:12]: 解决编译sample编译中的一个错误，我另外一个commit吧，掺和在一起不大合适的

**Review by yuanchaoa — APPROVED** [review]

**Pull request merged**